modules/centrifuge/main.nf

```diff
@@ -0,0 +1,38 @@
+process CENTRIFUGE {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::centrifuge=1.0.4_beta" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/centrifuge:1.0.4_beta--h9a82719_6' :
+        'quay.io/biocontainers/centrifuge:1.0.4_beta--h9a82719_6' }"
+
+    input:
+    tuple val(meta), path(reads)
+    path db
+
+    output:
+    tuple val(meta), path('*report.txt')  , emit: report
```

Comment:
This tool can also output SAM files, you will need to account for this here, e.g. with a new line

```
tuple val(meta), path('*.sam') , optional: true emit: sam
```
Based on
```
  --out-fmt <str>       define output format, either 'tab' or 'sam' (tab)
```